docs/integrations/prometheus.txt

```diff
@@ -0,0 +1,175 @@
+GPU metrics. Although other monitoring tools can be used with the setup,
+this guide describes only cAdvisor and DCGM tool configuration.
+Prometheus queries on metrics collected by other tools can differ,
+depending on the format and organization of the returned metrics.
+
+***************
+ Prerequisites
+***************
+
+-  A Grafana installation for dashboard monitoring.
+-  An on-cluster Prometheus instance for time-series data collection.
+
+**********************
+ Configure Determined
+**********************
+
+Install and run Determined on a cluster. When launching the master
+instance, enable the Prometheus endpoints by adding a flag to the
+``master.yaml`` configuration file:
+
+.. code:: yaml
+
+   observability:
+       enable_prometheus: true
+
+This enables the following two Prometheus API endpoints on the instance.
+
+-  ``{$DET_MASTER_ADDR}/prom/det-state-metrics``:
+
+   The ``det-state-metrics`` endpoint includes various machine-level
+   label mappings to internal Determined entities, such as GPU UUIDs and
+   container IDs to task, allocation, and experiment labels, which are
+   used by PromQL to join vectors.
+
+-  ``{$DET_MASTER_ADDR}/prom/det-http-sd-config``:
+
+   The ``det-http-sd-config`` endpoint contains address and resource
+   pool information for currently active agents. These are used by
+   Prometheus as targets for scraping. This endpoint is configured to
+   support running default cAdvisor, port 8080, and DCGM, port 9400,
+   monitoring. Other tools exposing Prometheus metrics can be used
+   instead of cAdvisor and DCGM if they are running on these ports.
+
+**************************************
+ Configure cAdvisor and dcgm-exporter
+**************************************
+
+The cAdvisor and dcgm-exporter tools must be running on the cluster
+agents to be monitored. These can be installed manually or run as
+individual Docker containers.
+
+To configure dynamic agents to start up with cAdvisor and dcgm-exporter,
+add the following startup script to the ``master.yaml`` file:
+
+.. code:: yaml
+
+   - pool_name: compute-pool
+       provider:
+         startup_script: |
+           # Run dcgm-exporter on 9400
+           docker run -d --gpus all --rm -p 9400:9400 nvcr.io/nvidia/k8s/dcgm-exporter:2.3.2-2.6.3-ubuntu20.04
+
+           # Run cAdvisor on 8080
+           VERSION=v0.36.0
+           docker run \
+             --volume=/:/rootfs:ro \
+             --volume=/var/run:/var/run:ro \
+             --volume=/sys:/sys:ro \
+             --volume=/var/lib/docker/:/var/lib/docker:ro \
+             --volume=/dev/disk/:/dev/disk:ro \
+             --publish=8080:8080 \
+             --detach=true \
+             --name=cadvisor \
+             --privileged \
+             --device=/dev/kmsg \
+             gcr.io/cadvisor/cadvisor:$VERSION
+
+This example startup script includes the default setup docker commands
+provided by dcgm-exporter and cAdvisor.
```

Comment:
not sure if I had the links in my latest revision, but can we include links here to dcgm-exporter and cAdvisor setup pages? these pages have the docker commands listed and may be useful if a user wants to dig deeper into these commands. 

[dcgm-exporter](https://github.com/NVIDIA/dcgm-exporter) and [cAdvisor](https://github.com/google/cadvisor).

Reply:
Since I added the links in 'Constraints' (the first mention) I removed them here. But I'll restore these as being only under 'Constraints' is too obscure.